tests/all/unit/api/validation/test_discount_codes.py

```diff
@@ -0,0 +1,53 @@
+import unittest
+
+from app import current_app as app
+from tests.all.integration.utils import OpenEventTestCase
+from app.api.helpers.exceptions import UnprocessableEntity
+from app.api.schema.discount_codes import DiscountCodeSchemaTicket
+from app.factories.ticket import TicketFactory
+from app.api.helpers.db import save_to_db
+
+
+class TestDiscountCodeValidation(OpenEventTestCase):
+
+    def test_percent_value_gt_hundred(self):
+        """
+        Discount Code Validate Percentage Value - Tests if exception is raised when percentage value is greater than 100
+        :return:
+        """
+        schema = DiscountCodeSchemaTicket()
+        original_data = {
+            'data': {}
+        }
+        data = {
+            'type': 'percent',
+            'value': 110
+        }
+        with self.assertRaises(UnprocessableEntity):
+            DiscountCodeSchemaTicket.validate_value(schema, data, original_data)
```

Comment:
This should be in unit test

Reply:
@iamareebjamal they are presen in `unit` test folder. If this is not what you wanted, can you please tell me the directory structure you wanted.

Reply:
You have written 2 tests. One is unit, other is integration. Move both to corresponding folders

Reply:
@iamareebjamal I've updated the files and added `test_amount_lte_ticket_price` and `test_amount_gt_ticket_price` to `integration` - `test_percent_value_lte_hundred` and `test_percent_value_gt_hundred` to `unit`. Is this fine now?

Reply:
It's fine, but no need to repeat test

Reply:
@iamareebjamal tests are not repeated. Actually one of the test will raise an exception and other will not. This will ensure covergae of complete tests written

Reply:
Yes but the same two tests are repeated in unit tests as well, one that raises exception and other that does not. What's the point of that?

Reply:
Actually what I'm trying to do is to cover entire codebase of `validate_value` function to get more coverage.

Reply:
@iamareebjamal do u want me to remove the tests which pass without throwing any exception?

Reply:
There are 4 functions. 2 check if validate throws, 2 check if it doesn't throw. Tell me why there aren't just 2 functions. One which throws and one which doesn't

Reply:
OK, got it. 2 are for percent and 2 are for amount. Why can't percent test be unit as well?

Reply:
It can be in unit as well. Actually I shifted all the 4 tests to unit tests section but asked me to move it to `integration` tests again :sweat_smile: 
I get it, there was a confusion. Should I place all the 4 tests to unit tests now? I think it'll be better this way and this PR will be finalized hopefully @iamareebjamal

Reply:
Yes, you can move them but I asked you to move them to integration because they are using app.context and saving to db, which cannot be done using unit test. So the question is not about just moving the tests. Why do you need DB? Why can't the percentage test be just a unit test like the amount tests?

Reply:
@iamareebjamal I'm getting confused here :sweat_smile:  -
1. The percent tests are already in `unit` test section and also do not use db
2. Db is used by `amount` tests and they are in `integration` section. 
3. It is necessary to use db for `amount` test because the discount codes need to be associated with tickets for this test. So, I saved a ticket to db and passed its id to `data`.

Reply:
OK. Got it :+1: 

Please handle negative amount and percent as well

Reply:
@iamareebjamal I've added checks for negative amount and percent values. Should I add tests for these conditions too?